finality-aleph/src/party/mod.rs

```diff
@@ -483,14 +544,17 @@ where
                         None => None,
                     } } => {
                     warn!(target: "afa", "Authority task ended prematurely, restarting.");
```

Comment:
When do we expect this to happen?

Reply:
When something will die, for example due to lack of public address